packages/embark/src/lib/modules/tests/test.js

```diff
@@ -331,15 +327,39 @@ class Test {
 
             const testContractFactoryPlugin = self.plugins.getPluginsFor('testContractFactory').slice(-1)[0];
 
-            const newContract = testContractFactoryPlugin ? testContractFactoryPlugin.testContractFactory(contract, web3) : Test.getWeb3Contract(contract, web3);
-            Object.setPrototypeOf(self.contracts[contract.className], newContract);
-
-            eachCb();
+            if (testContractFactoryPlugin) {
+              const newContract = testContractFactoryPlugin.testContractFactory(contract, web3);
+              Object.setPrototypeOf(self.contracts[contract.className], newContract);
+              return eachCb();
+            }
+            self.getEmbarkJSContract(contract, (err, vmContract) => {
+              if(err) {
+                self.logger.error(`Erroring creating contract instance '${contract.className}' for import in to test. Error: ${err}`);
```

Comment:
When this tiggers, `vmContract` will be undefined, probably thowing at the `Object.setPrototypeOf`
Probably needs a `return eachCb(err);`

Reply:
`vmContract || null` will not throw.